### PR TITLE
Comments and Documentation

### PR DESCRIPTION
**Pull request by hamza-sadruddin**
Please have a look at the following files for Task C,

- Readme.md
- receiver_cadmium.hpp (Hamza)
- subnet_cadmium.hpp (Hamza)

### CODE REVIEW

**Review by abhinavachanta — COMMENTED** [review]
These files looks good. Waiting for other 2 files to be uploaded.

**Comment by khanyousefzai** [comment]
Please check comments on the file main.cpp and message.cpp.
Thank You

File: include/data_structures/message.hpp [hunk 1]
```diff
@@ -28,4 +33,4 @@ istream& operator>> (istream& is, message_t& msg);
 ostream& operator<<(ostream& os, const message_t& msg);
 
 
-#endif // BOOST_SIMULATION_MESSAGE_HPP
\ No newline at end of file
+#endif // < BOOST_SIMULATION_MESSAGE_HPP - ifndef macro end
```
Comment (jrajeev459) [inline include/data_structures/message.hpp]: Missed Underscore in here  BOOST_SIMULATION_MESSAGE_HPP

File: include/receiver_cadmium.hpp [hunk 2]
```diff
@@ -28,42 +39,68 @@
...
-    using defs = receiver_defs; // putting definitions in context
+    /** Putting definitions in context*/
+    using defs = receiver_defs; 
     public:
-        //Parameters to be overwriten when instantiating the atomic model
-        TIME PREPARATION_TIME;
-        // default constructor
+        
+        TIME PREPARATION_TIME; /**<Time delay constant holder from acknowledgement to output. */
+                               //!< Time constant delay
```
Comment (jrajeev459) [inline include/receiver_cadmium.hpp]: Follow consistency of the comment style.

File: include/receiver_cadmium.hpp [hunk 3]
```diff
@@ -28,42 +39,68 @@
...
         
-        // state definition
+        /**
+         * Structure state definition that holds acknowledge number and receiver state.
+         */
         struct state_type{
-            int ack_num;
-            bool sending;
+            int ack_num;    /**< Alternating bit fetched from the message and acknowledgement number. */
+                            //!< Acknowledgement number
```
Comment (jrajeev459) [inline include/receiver_cadmium.hpp]: Follow consistency of the comment style.

File: include/sender_cadmium.hpp [hunk 4]
```diff
@@ -178,4 +178,4 @@ class Sender{
 };     
 
 
-#endif // __BOOST_SIMULATION_PDEVS_SENDER_HPP__
\ No newline at end of file
+#endif // __SENDER_CADMIUM_HPP__
```
Comment (jrajeev459) [inline include/sender_cadmium.hpp:181]: Follow consistency of the comment style.

**Comment by jrajeev459** [comment]
please check the above comments made.

**Comment by hamza-sadruddin** [comment]
> please check the above comments made.

I don't see any comments you have made.

File: include/receiver_cadmium.hpp (hunk 3 above)
Comment (hamza-sadruddin) [reply include/receiver_cadmium.hpp]: Done. Please have a look at it again.

File: include/receiver_cadmium.hpp (hunk 2 above)
Comment (hamza-sadruddin) [reply include/receiver_cadmium.hpp]: Done. Please have a look at it again.

File: include/sender_cadmium.hpp (hunk 4 above)
Comment (hamza-sadruddin) [reply include/sender_cadmium.hpp:181]: We are not modifying this file.

Comment (jrajeev459) [reply include/sender_cadmium.hpp:181]: okay.

**Review by jrajeev459 — APPROVED** [review]

File: src/message.cpp [hunk 5]
```diff
@@ -6,19 +6,23 @@
 
 #include "../include/data_structures/message.hpp"
```
Comment (abhinavachanta) [inline src/message.cpp:14]: Also add a comment on starting of this file (on the top). Use 'brief' word to give a summary of what this file does

**Review by abhinavachanta — CHANGES_REQUESTED** [review]
Also add a comment on starting of this file 'src/message.cpp'. Use 'brief' word to give a summary of what this file does

File: include/data_structures/message.hpp (hunk 1 above)
Comment (khanyousefzai) [reply include/data_structures/message.hpp]: Done. Please check again. Thanks

File: src/message.cpp (hunk 5 above)
Comment (khanyousefzai) [reply src/message.cpp:14]: Done. Please check again. Thanks

File: README.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    - message.hpp
+- receiverCadmium.hpp
+- senderCadmium.hpp
+- subnetCadmium.hpp
+
+**lib [This folder contains 3rd party header files needed in the project]**
+1. cadmium [This folder contains cadmium functionalities]
+2. DESTimes [This folder contains DESTimes functionalities]
+-	iestream.hpp
+-	NDTime.hpp
```
Comment (abhinavachanta) [inline README.md:54]: Remove the NDTime.hpp file from lib directory, and use the NDTime.hpp file directly from DESTimes/NDTime.hpp as professor mentioned. Accordingly update all paths where NDTiime.hpp is imported in src and test files.

**Review by abhinavachanta — CHANGES_REQUESTED** [review]
Remove the NDTime.hpp file from lib directory, and use the NDTime.hpp file directly from DESTimes/NDTime.hpp as professor mentioned. Accordingly, update all paths where NDTiime.hpp is imported in src and test files.

File: README.md (hunk 6 above)
Comment (hamza-sadruddin) [reply README.md:54]: Yes, that's part of Task-D(a). You are mistaken. Please have a look at it again. Thanks.

Comment (abhinavachanta) [reply README.md:54]: Yes. Sorry, I overlooked it as part-D. 
Part-c comments look good.

**Review by abhinavachanta — APPROVED** [review]
Comments added by developers as per the standards.

**Pull request merged**